core/templates/pages/exploration-editor-page/services/change-list.service.ts

```diff
@@ -17,246 +17,280 @@
  * committed to the server.
  */
 
-import { EventEmitter } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { EventEmitter, OnInit, Output } from '@angular/core';
+import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
-require(
-  'pages/exploration-editor-page/services/autosave-info-modals.service.ts');
-require('pages/exploration-editor-page/services/exploration-data.service.ts');
-require('services/alerts.service.ts');
-
-angular.module('oppia').factory('ChangeListService', [
-  '$log', '$rootScope', 'AlertsService', 'AutosaveInfoModalsService',
-  'ExplorationDataService', 'LoaderService',
-  function(
-      $log, $rootScope, AlertsService, AutosaveInfoModalsService,
-      ExplorationDataService, LoaderService) {
-    // TODO(sll): Implement undo, redo functionality. Show a message on each
-    // step saying what the step is doing.
-    // TODO(sll): Allow the user to view the list of changes made so far, as
-    // well as the list of changes in the undo stack.
-
-    // Temporary buffer for changes made to the exploration.
-    var explorationChangeList = [];
-    // Stack for storing undone changes. The last element is the most recently
-    // undone change.
-    var undoneChangeStack = [];
-    var loadingMessage = '';
-    LoaderService.onLoadingMessageChange.subscribe(
-      (message: string) => loadingMessage = message
+
+import { AutosaveInfoModalsService } from 'pages/exploration-editor-page/services/autosave-info-modals.service.ts';
+import { ExplorationDataService } from 'pages/exploration-editor-page/services/exploration-data.service';
+import { AlertsService } from 'services/alerts.service';
+import { LoaderService } from 'services/loader.service';
+import { LoggerService } from 'services/contextual/logger.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ChangeListService implements OnInit {
+  // TODO(sll): Implement undo, redo functionality. Show a message on each
+  // step saying what the step is doing.
+  // TODO(sll): Allow the user to view the list of changes made so far, as
+  // well as the list of changes in the undo stack.
+
+  // Temporary buffer for changes made to the exploration.
+  explorationChangeList: Object[] = [];
```

Comment:
Not Object. This should be ExplorationChange[] I think

Reply:
done!